src/main/java/gyro/google/compute/ProjectMetadataItemResource.java

```diff
@@ -0,0 +1,174 @@
+        return value;
+    }
+
+    public void setValue(String value) {
+        this.value = value;
+    }
+
+    @Override
+    public void copyFrom(Metadata.Items metadataItem) {
+        setKey(metadataItem.getKey());
+        setValue(metadataItem.getValue());
+    }
+
+    @Override
+    public boolean refresh() {
+        Compute client = creatClient(Compute.class);
+
+        Metadata metadata = getMetadata(client);
+        Metadata.Items item = metadata.getItems()
+            .stream()
+            .filter(r -> getKey().equals(r.getKey()))
+            .findFirst()
+            .orElse(null);
+
+        if (item != null) {
+            copyFrom(item);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public void create(GyroUI ui, State state) {
+        Compute client = creatClient(Compute.class);
+
+        Metadata.Items item = new Metadata.Items();
+        item.setKey(getKey());
+        item.setValue(getValue());
+
+        Metadata metadata = getMetadata(client);
+        List<Metadata.Items> items = metadata.getItems();
+        items.add(item);
+        metadata.setItems(items);
+
+        setMetadata(client, metadata);
+    }
+
+    @Override
+    public void update(GyroUI ui, State state, Resource current, Set<String> changedFieldNames) {
+        Compute client = creatClient(Compute.class);
+
+        Metadata metadata = getMetadata(client);
+        Metadata.Items item = metadata.getItems()
+            .stream()
+            .filter(r -> getKey().equals(r.getKey()))
+            .findFirst()
+            .orElse(null);
+
+        if (item != null) {
+            item.setValue(getValue());
+            setMetadata(client, metadata);
+        }
+    }
+
+    @Override
+    public void delete(GyroUI ui, State state) {
+        Compute client = creatClient(Compute.class);
+
+        Metadata metadata = getMetadata(client);
+        metadata.getItems().removeIf(r -> getKey().equals(r.getKey()));
+
+        setMetadata(client, metadata);
+    }
+
+    private Metadata getMetadata(Compute client) {
+        try {
+            Compute.Projects.Get projectRequest = client.projects().get(getProjectId());
+            Project project = projectRequest.execute();
+
+            return project.getCommonInstanceMetadata();
+        } catch (Exception ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    private void setMetadata(Compute client, Metadata metadata) {
+        try {
+            Compute.Projects.SetCommonInstanceMetadata metadataRequest =
+                client.projects().setCommonInstanceMetadata(getProjectId(), metadata);
+            Operation operation = metadataRequest.execute();
+
+            Operation.Error error = waitForCompletion(client, operation);
+            if (error != null) {
+                throw new GyroException(error.toPrettyString());
+            }
+        } catch (GoogleJsonResponseException je) {
```

Comment:
Currently for duplicate keys the error response looks like
```
Invalid value for field 'resource': '{  "fingerprint": "KjSjVaX4e8I=",  "item": [{    "key": "ssh-keys",    "value": "deepanjan:ssh-rsa A...'. Metadata has duplicate keys: ssh-keys
```

For this particular case we could know that its a duplicate key, and throw a proper formatter error
Something along the lines of `Duplicate key: <Key_Name>`